feature_detection/scripts/Hough_Transform_orientation_based.py

```diff
@@ -0,0 +1,301 @@
+import cv2
+import numpy as np
+from copy import deepcopy
+
+
+''' 
+ToDo
++ tuning of the parameter of cv2.HoughLinesP makes it more robust, but less sensitive
++ component detection --> some kind of filter(?) has to be applied
++ idea on how to improve the algorithm: use eucledian norm to get distance of two lines to get independent of orientation
+
+'''
+
+# feature detection based on hough transform
+class HoughMajingo_ob:
+    """
+    Class used in feature detection to detect shapes of objects consisting of straight lines. Returns the bounding boxes and the centroids of them
+
+    """
+
+    @staticmethod
+    def lines_coord(lines_HT, set, e=8):
+        '''
+        Since the image is affected by noise and calibration inaccuracies, this function attempts to make the output of the bounding boxes more robust 
+        by combining adjacent lines into one line. 
+        
+        args: 
+            lines_HT    : List of start and end coordinates of lines created by the HougTransform method 
+                        and containing either horizontal or vertical lines.
+            set         : parameter to adapt algorithm to either horizontal or vertical lines.
+                        set = 0: vertical lines
+                        set = 1: horizontal lines
+            e           : {default=8} adjacent lines, which have a distance less than margin e, are represented by a line by averaging them
+
+        return:
+            rect_list   : returns starting and end point of detected line that satisfy the margin
+            position    : returns the mean of the "position" of the line 
+                        -> vertical line: x_position, horizontal line: y_position
+        '''
+        
+        if set == 0:
+            ver = 1  # value for vertical lines
+        elif set ==1:
+            ver = 0  # value for horizontal lines
```

Comment:
Switch the meaning of "set" so we can instead do

```suggestion
        def lines_coord(lines_HT, set=0, e=8):
            ...
            if not set in [0, 1]:
                # Do something to handle this
            ver = set
```